File: server/src/main/java/xyz/chunkstories/server/commands/debug/MiscDebugCommands.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.server.commands.debug;

import xyz.chunkstories.api.plugin.commands.Command;
import xyz.chunkstories.api.plugin.commands.CommandEmitter;
import xyz.chunkstories.api.server.Server;
import xyz.chunkstories.server.commands.ServerCommandBasic;

public class MiscDebugCommands extends ServerCommandBasic {

	public MiscDebugCommands(Server serverConsole) {
		super(serverConsole);
		server.getPluginManager().registerCommand("gc", this);
	}

	@Override
	public boolean handleCommand(CommandEmitter emitter, Command command, String[] arguments) {
		if (command.getName().equals("gc") && emitter.hasPermission("server.debug.gc")) {
			emitter.sendMessage("Performing gc...");
			System.gc();
			return true;
		}
		return false;
	}

}

File: server/src/main/java/xyz/chunkstories/server/commands/debug/MiscDebugCommands.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.server.commands.debug

import xyz.chunkstories.api.plugin.commands.Command
import xyz.chunkstories.api.plugin.commands.CommandEmitter
import xyz.chunkstories.api.server.Host
import xyz.chunkstories.server.commands.AbstractHostCommandHandler

class MiscDebugCommands(serverConsole: Host) : AbstractHostCommandHandler(serverConsole) {

    init {
        host.pluginManager.registerCommand("gc", this)
    }

    override fun handleCommand(emitter: CommandEmitter, command: Command, arguments: Array<String>): Boolean {
        if (command.name == "gc" && emitter.hasPermission("server.debug.gc")) {
            emitter.sendMessage("Performing gc...")
            System.gc()
            return true
        }
        return false
    }

}
